src/semidiscretization/semidiscretization_coupled_p4est.jl

```diff
@@ -0,0 +1,544 @@
+                                                             cache,
+                                                             i_index, j_index,
+                                                             element_index,
+                                                             normal_direction,
+                                                             surface_flux_function,
+                                                             direction,
+                                                             u_global)
+    n_nodes = length(mesh.parent.nodes)
+    if abs(sum(normal_direction .* (1.0, 0.0))) >
+       abs(sum(normal_direction .* (0.0, 1.0)))
+        if sum(normal_direction .* (1.0, 0.0)) > sum(normal_direction .* (-1.0, 0.0))
+            element_index_global = cache.neighbor_ids_global[findfirst((cache.boundaries.name .==
+                                                                        :x_pos) .*
+                                                                       (cache.boundaries.neighbor_ids .==
+                                                                        element_index))]
+        else
+            element_index_global = cache.neighbor_ids_global[findfirst((cache.boundaries.name .==
+                                                                        :x_neg) .*
+                                                                       (cache.boundaries.neighbor_ids .==
+                                                                        element_index))]
+        end
+        i_index_g = i_index
+        if i_index == n_nodes
+            i_index_g = 1
+        elseif i_index == 1
+            i_index_g = n_nodes
+        end
+        j_index_g = j_index
+    else
+        if sum(normal_direction .* (0.0, 1.0)) > sum(normal_direction .* (0.0, -1.0))
+            element_index_global = cache.neighbor_ids_global[findfirst((cache.boundaries.name .==
+                                                                        :y_pos) .*
+                                                                       (cache.boundaries.neighbor_ids .==
+                                                                        element_index))]
+        else
+            element_index_global = cache.neighbor_ids_global[findfirst((cache.boundaries.name .==
+                                                                        :y_neg) .*
+                                                                       (cache.boundaries.neighbor_ids .==
+                                                                        element_index))]
+        end
+        j_index_g = j_index
+        if j_index == n_nodes
+            j_index_g = 1
+        elseif j_index == 1
+            j_index_g = n_nodes
+        end
+        i_index_g = i_index
+    end
+    u_global_reshape = reshape(u_global,
+                               (n_nodes, n_nodes, length(u_global) ÷ n_nodes^2))
```

Comment:
Maybe add a comment why [integer division](https://docs.julialang.org/en/v1/manual/mathematical-operations/#Arithmetic-Operators) is sufficient here?